enhancements/ingress/set-delete-http-headers.md

```diff
@@ -1123,6 +1159,33 @@ We have multiple options for quoting header values in `haproxy.config`, as [desc
   2. Wrap the entire string in single-quotes.
 The HAProxy documentation refers to this as _strong quoting_.  Using strong quoting allows us to keep the quoting logic simple and `haproxy.config` more human-readable compared to the alternatives.
 
+### Proposed Alternative Solution for overriding the HTTP response headers defined in the IngressController by the Route
+As per HAProxy's design: the frontend receives the request and dispatches it to the backend, and then the frontend
+gets the response from the backend.
+
+So, the override behavior happens the way mentioned in the question `Why does a custom header value defined in RouteSpec override IngressControllerSpec value ?`
+under section `Open Questions` where HTTP request headers defined in the IngressController are overridden by headers 
+defined in the Route.
+
+However, if we want the same behavior of override for HTTP response headers as that of HTTP request headers then we will have
+to inject the set-header from the IngressController config into every backend and not the frontend.
+
+The downside is that it would grow the generated haproxy.config significantly. 
+For Example: Suppose an ingress controller called custom-header handles 100 routes.
+If it sets HTTP response X-Frame-Options DENY via spec.httpHeader.actions, 
+then for all those 100 routes `http-response set-header X-Frame-Options DENY` will be set i.e. 100 stanzas as it will be one stanza per backend related to a route. 
+
+There might be a case, where admin wants to set a response header for all connections to the IngressController as per company policy which he won't be able to do so
+using this approach.
+For Example: A cluster admin wants to set the  `X-Frame-Options` header with the value `DENY` for all routes the ingress controller is handling, overwriting 
+any existing `X-Frame-Options` header that may have been set by the application or Route configuration.
+
+The upside of putting all the header configuration on the backends is that
+it would give OpenShift router full control over the ordering of header actions.
```

Comment:
```suggestion
The upsides of injecting all the header configuration on the backends is that
it would give OpenShift router full control over the ordering of header actions and present a more consistent learning environment for the admin.
```

Reply:
ok done